src/any_llm/providers/bedrock/utils.py

```
@@ -38,25 +46,118 @@
-        raise UnsupportedParameterError(
-            msg,
-            "bedrock",
-            "Check the following links:\n- https://docs.aws.amazon.com/nova/latest/userguide/prompting-structured-output.html",
-        )
-
-    if params.tools:
-        result_kwargs["toolConfig"] = _convert_tool_spec(params.tools, params.tool_choice)
+    _check_no_reserved_tool_name_collision(params.tools)
 
     reasoning_enabled = (
         params.reasoning_effort is not None and params.reasoning_effort != "auto" and params.reasoning_effort != "none"
```

Ruff (0.16.1)

[warning] Consider merging multiple comparisons: `params.reasoning_effort not in {"auto", "none"}`.

Merge multiple comparisons

(PLR1714)